src/effects/VST3/VST3Effect.cpp

```diff
@@ -339,6 +339,68 @@ bool VST3Effect::SupportsAutomation() const
    return false;
 }
 
+
+bool VST3Effect::FetchSettings(VST3EffectSettings& settings) const
+{
+   // partly copied from ::SaveSettings + some adaptations made
+
+   if (mEditController == nullptr)
+      return false;
+
+   using namespace Steinberg;
+
+   auto theCount = mEditController->getParameterCount();
+
+   for (int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
+   {
+      Vst::ParameterInfo parameterInfo{ };
+      if (mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+      {
+         if (parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
```

Comment:
Parameters stored by `SaveSettings` are used by macroses in Audacity and are meant to be read by human, I'm not sure if that was correct to skip other types of parameters, but this was an attempt to restrict the amount of parameters (there could be thousands of them) to a set of paramters that are useful and meaningful

Reply:
Can we assume that the set of parameters, and the kCanAutomate flag for each parameter, is unchanging during the lifetime of the loaded VST3 plug-in?  And independent of any instance state?  I would assume so.

When you write CopySettingsContents analogous to that for AudioUnits (not yet merged, but in the last of the sequence of PRs for AU), you may be making such an assumption.